snowplow_tracker/events/event.py

```diff
@@ -0,0 +1,134 @@
+# """
+#     event.py
+
+#     Copyright (c) 2013-2023 Snowplow Analytics Ltd. All rights reserved.
+
+#     This program is licensed to you under the Apache License Version 2.0,
+#     and you may not use this file except in compliance with the Apache License
+#     Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+#     http://www.apache.org/licenses/LICENSE-2.0.
+
+#     Unless required by applicable law or agreed to in writing,
+#     software distributed under the Apache License Version 2.0 is distributed on
+#     an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#     express or implied. See the Apache License Version 2.0 for the specific
+#     language governing permissions and limitations there under.
+# """
+
+from typing import Optional, List
+from snowplow_tracker import payload
+from snowplow_tracker.subject import Subject
+
+from snowplow_tracker.self_describing_json import SelfDescribingJson
+
+from snowplow_tracker.constants import CONTEXT_SCHEMA
+from snowplow_tracker.typing import JsonEncoderFunction, PayloadDict
+
+
+class Event(object):
+    """
+    Event class which contains
+    elements that can be set in all events. These are context, trueTimestamp, and Subject.
+
+    Context is a list of custom SelfDescribingJson entities.
+    TrueTimestamp is a user-defined timestamp.
+    Subject is an event-specific Subject. Its fields will override those of the
+    Tracker-associated Subject, if present.
+
+    """
+
+    def __init__(
+        self,
+        dict_: Optional[PayloadDict] = None,
+        event_subject: Optional[Subject] = None,
+        context: Optional[List[SelfDescribingJson]] = None,
+        true_timestamp: Optional[float] = None,
+    ) -> None:
+        """
+        Constructor
+        :param  dict_:              Optional Dictionary to be added to the Events Payload
+        :type   dict_:              dict(string:\\*) | None
+        :param  event_subject:      Optional per event subject
+        :type   event_subject:      subject | None
+        :param  context:            Custom context for the event
+        :type   context:            context_array | None
+        :param  true_timestamp:     Optional event timestamp in milliseconds
+        :type   true_timestamp:     int | float | None
+
+        """
+        self.payload = payload.Payload(dict_=dict_)
+        self.event_subject = event_subject
+        self.context = context
```

Comment:
It would be great if we initialized the context even if it is not passed in so that users can add entities to it:

```suggestion
        self.context = context or []
```

In that case we can update the `context` property below so that it is not optional.